evalml/utils/gen_utils.py

```diff
@@ -379,3 +380,98 @@ def _not_nan(pd_data):
 
     mask = np.logical_and(_not_nan(pd_data_1), _not_nan(pd_data_2))
     return pd_data_1.iloc[mask], pd_data_2.iloc[mask]
+
+
+def _file_path_check(filepath=None, format='png', interactive=False, is_plotly=False):
+    """ Helper function to check the filepath being passed.
+
+    Arguments:
+        filepath (str or Path, optional): Location to save file.
+        format (str): Extension for figure to be saved as. Defaults to 'png'.
+        interactive (bool, optional): If True and fig is of type plotly.Figure, sets the format to 'html'.
+        is_plotly (bool, optional): Check to see if the fig being passed is of type plotly.Figure.
+
+    Returns:
+        String representing the final filepath the image will be saved to.
+    """
+    if filepath:
+        filepath = str(filepath)
+        path_and_name, extension = os.path.splitext(filepath)
+        extension = extension[1:].lower() if extension else None
```

Comment:
Although it is in the docstring, and users can choose to return the final filepath, I wanted to bring up a suggestion to somehow give a message to the user about the file saving under a different name than the user expected. 

For instance, if the user passed in `save_plot(fig, filepath="plot.png", format="png", interactive=True)` and fig was a plotly.Figure, it could save some confusion if we told them something along the lines of `Saving plot as 'plot.html' since 'interactive=True' and fig is a plotly.Figure`. 

Definitely not necessary since it's all in the docstring already, but I figure it could save some confusion, so I wanted to bring it up and see what other people thought!

Reply:
@bchen1116 Good point, and just for clarity I will add the specification that the format would be set to html in the docstring.